src/Faction/Faction.ts

```diff
@@ -34,6 +35,37 @@ export class Faction {
     this.name = name;
   }
 
+  get favor() {
+    return this.#favor;
+  }
+
+  /**
+   * There is no setter for this.#favor. This is intentional. Performing arithmetic operations on `favor` may lead to
+   * the overflow error of `playerReputation`, so anything that wants to change `favor` must explicitly do that through
+   * `setFavor` and `gainFavor`.
+   *
+   * Only use this function in these cases:
+   * - Initialize favor = 0.
+   * - Set value returned from calculateFavorAfterResetting function.
+   * - Load value from save data.
+   * - Dev menu.
+   *
+   * If you want to increase the favor, you must use the `gainFavor` function.
```

Comment:
I see the justification here, but I don't agree with it. *Adding* favor is perfectly fine; the clamp in setFavor deals with that. The potential issue comes in if you are doing more complicated math that might divide infinity by infinity, leading to NaN. But those cases can't be handled with gainFavor anyway; they were always going to have to do calculations and then set the value. 

Basically, although I understand your motivation, I think you are only protecting the case that was already safe.

Reply:
Addendum: I can understand intentionally wanting to add some friction so that people don't assume this is a "normal" variable. For that, possibly not having a normal setter (keep it explicit), but drop gainFavor since it doesn't add anything?